### PR TITLE
Use POWER USB_DETECTED IRQ instead of detecting VBUS_ABSENT via GPIOTE

### PR DESCRIPTION
**Pull request by martinwork**
Includes macros to switch between USBDETECTED and VBUS_ABSENT

### CODE REVIEW

File: source/board/microbitv2/microbitv2.c [hunk 1]
```diff
@@ -202,6 +202,14 @@ static inline uint8_t get_led_gamma(uint8_t brightness) {
...
 static void prerun_board_config(void)
 {
+#if defined(INTERFACE_NRF52820)
+    if (NRF_POWER->RESETREAS & POWER_RESETREAS_VBUS_Msk) {
+        wake_from_usb = true;
+    }
+    if (NRF_POWER->RESETREAS & POWER_RESETREAS_OFF_Msk) {
+        wake_from_reset = true;
+    }
+#endif
```
Comment (microbit-carlos) [inline source/board/microbitv2/microbitv2.c]: Can this be moved to a function from power.h (or done in `power_init()`) to reduce the number of #if defines in common files to both MCUs?

File: source/board/microbitv2/microbitv2.c [hunk 2]
```diff
@@ -244,7 +252,7 @@ void handle_reset_button()
     {
         // handle reset button without eventing
         if (!reset_pressed && (gpio_get_reset_btn_fwrd() || wake_from_reset)) {
-            // Reset button pressed
+             // Reset button pressed
```
Comment (microbit-carlos) [inline source/board/microbitv2/microbitv2.c]: We can revert this

File: source/board/microbitv2/nrf52820/power.c [hunk 3]
```diff
@@ -57,6 +57,9 @@ extern power_source_t power_source;
 uint8_t  power_gpiote_enabled;
 uint32_t power_gpiote_intenset;
 
+// Define one of these
```
Comment (microbit-carlos) [inline source/board/microbitv2/nrf52820/power.c:60]: Might be good to have a short description about the difference between the two options and why we are using USBDETECTED instead

File: source/board/microbitv2/nrf52820/power.c [hunk 4]
```diff
@@ -216,5 +249,28 @@ void GPIOTE_IRQHandler(void)
...
+        NRF_POWER->EVENTS_USBREMOVED = 0;
+        power_source = pwr_mon_get_power_source();
+        /* Reset USB on cable detach (VBUS falling edge) */
+        USBD_Reset();
+        usbd_reset_core();
+        usb_pc_connected = false;
+        usb_state = USB_DISCONNECTED;
     }
 }
+#endif // POWER_IRQ_USBDETECTED
```
Comment (microbit-carlos) [inline source/board/microbitv2/nrf52820/power.c]: New line at the end of the file

**Review by microbit-carlos — COMMENTED** [review]
Great, let's go with POWER_IRQ_USBDETECTED 👍

File: source/board/microbitv2/microbitv2.c (hunk 1 above)
Comment (martinwork) [reply source/board/microbitv2/microbitv2.c]: Oh dear, yes, absolutely right! Thanks.

File: source/board/microbitv2/microbitv2.c (hunk 2 above)
Comment (martinwork) [reply source/board/microbitv2/microbitv2.c]: Done

File: source/board/microbitv2/nrf52820/power.c (hunk 3 above)
Comment (martinwork) [reply source/board/microbitv2/nrf52820/power.c:60]: Added

File: source/board/microbitv2/nrf52820/power.c (hunk 4 above)
Comment (martinwork) [reply source/board/microbitv2/nrf52820/power.c]: Done

**Review by microbit-carlos — APPROVED** [review]
Awesome, thanks Martin!

**Pull request merged**